docs/extending.md

```diff
@@ -1,208 +1,296 @@
 
 1. Generating `shadowOf` methods for each of your shadows.
-2. Generating a ServiceLoader so your custom shadows are automatically applied if found on the classpath
-3. Invoking any `static` `@Resetter` methods on teardown to enable you to reset static state.
+2. Generating a `ServiceLoader` so your custom shadows are automatically applied if found on the classpath.
+3. Invoking any `static` [`@Resetter`](javadoc/latest/org/robolectric/annotation/Resetter.html) methods on teardown to enable you to reset static state.
 4. Perform additional validation and checking on your shadows.
 
-```groovy
-android {
-    defaultConfig {
+=== "Groovy"
+
+    ```groovy
+    android {
+      defaultConfig {
         javaCompileOptions {
-            annotationProcessorOptions {
-                className 'org.robolectric.annotation.processing.RobolectricProcessor'
-                arguments = [ 'org.robolectric.annotation.processing.shadowPackage' : 'com.example.myshadowpackage' ]
-            }
+          annotationProcessorOptions {
+            className 'org.robolectric.annotation.processing.RobolectricProcessor'
+            arguments = [ 'org.robolectric.annotation.processing.shadowPackage': 'com.example.myshadowpackage' ]
+          }
         }
+      }
     }
 
-}
+    dependencies {
+      annotationProcessor 'org.robolectric:processor:{{ robolectric.version.current }}'
+    }
+    ```
 
-dependencies {
-    annotationProcessor 'org.robolectric:processor:<version>'
-    ...
-}
-```
+=== "Kotlin"
 
-When you write your shadows in Kotlin, configure kapt:
+    When you write your shadows in Kotlin, configure [`kapt`](https://kotlinlang.org/docs/kapt.html):
 
-```groovy
-plugins {
-    id 'kotlin-kapt'
-}
+    ```kotlin
+    plugins {
+      id("kotlin-kapt")
+    }
 
-kapt {
-    arguments {
-        arg('org.robolectric.annotation.processing.shadowPackage', 'com.example.myshadowpackage')
+    kapt {
+      arguments {
+        arg("org.robolectric.annotation.processing.shadowPackage", "com.example.myshadowpackage")
+      }
     }
-}
 
-dependencies {
-    kapt  'org.robolectric:processor:<version>'
-    ...
-}
-```
+    dependencies {
+      kapt("org.robolectric:processor:{{ robolectric.version.current }}")
+    }
+    ```
 
 ## Best practices
 
-### Limit API surface area of shadows.
+### Limit API surface area of shadows
+
 Since [Robolectric 3.7](https://github.com/robolectric/robolectric/releases/tag/robolectric-3.7) `@Implementation` methods including `__constructor__` methods can be made `protected`. This is desirable as test code has no business calling these methods, by making your `@Implementation` methods protected you encourage test writers to call the public Android APIs instead.
 
-### Don't use `useinheritImplementationMethods`
-This is usually unnecessary and will be removed in [Robolectric 3.8](https://github.com/robolectric/robolectric/releases/tag/robolectric-3.8)
```

Comment:
`useinheritImplementationMethods` doesn't seem to exist anymore (as documented), and Robolectric 3.8 is a rather old version now. So I've removed that section.